### PR TITLE
Adding cache clearing to application template delete scenario

### PR DESCRIPTION
**Pull request by somindatommy**
Resolves: wso2/product-is#10155

### CODE REVIEW

**Comment by somindatommy** [comment]
PR builder passed: https://github.com/wso2/product-is/actions/runs/357626798

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java [hunk 1]
```diff
@@ -299,67 +302,31 @@ private Resource getResourceFromCacheByName(String resourceName, int tenantId)
...
-            ResourceByIdCacheKey resourceByIdCacheKey = new ResourceByIdCacheKey(resourceId, tenantDomain);
-            ResourceCacheEntry resourceCacheEntry = resourceByIdCache.getValueFromCache(resourceByIdCacheKey);
-            if (resourceCacheEntry != null) {
-                if (log.isDebugEnabled()) {
-                    String message = String.format("Entry found from Resource by id cache. Resource id: %s., Tenant " +
-                            "domain %s", resourceId, tenantDomain);
-                    log.debug(message);
-                }
-                return resourceCacheEntry.getResource();
+       ResourceByIdCacheKey resourceByIdCacheKey = new ResourceByIdCacheKey(resourceId, tenantDomain);
```
Comment (IsuraD) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java:305]: formatting issues

Comment (somindatommy) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java]: Fixed: 7618b179bb89b5d3b1354d4734a7136fad1637b5

**Review by IsuraD — APPROVED** [review]

**Comment by somindatommy** [comment]
PR builder passed https://github.com/wso2/product-is/actions/runs/358034375

**Pull request merged**